### ISSUE TITLE
Improve tests on examples/rust/*

### ISSUE THREAD

**Issue by neil2468** [build-trust/ockam#3005]
Here are some opportunities...

1. The tests in `examples/rust/get_started` do not cover significant portions of code. 
2. Some tests in `examples/rust/*` could still clash on TCP ports when run in parallel. Maybe they should ask the OS for free ports, rather then using hardcoded values (could use [portpicker](https://crates.io/crates/portpicker)?). 
3. `example_test_helper` could allow child processes to inherit environment variables from their parent. This _might_ allow child `cargo` processes to use the mold linker in CI.
4. Maybe tests in `examples/rust/*` should use the the `cargo` `--locked` argument as is done in other parts of the Ockam code base (e.g. [here](https://github.com/build-trust/ockam/blob/f6eb4126cd4ac5f7c265118ac580b70f6eea05fb/implementations/rust/build.gradle#L88)).

---

<i>We love helping new contributors! ❤️
If you have questions or need help as you explore, please join us on [Discord](https://discord.gg/bewvnm6zqS). If you're looking for other issues to contribute to, please checkout our [good first issues](https://github.com/build-trust/ockam/labels/good%20first%20issue).</i>

**Comment by PsychoPunkSage** [comment]
Hi @neil2468 
Can I work on this issue. it seems no one is working on it. Can I have a look at this?

**Comment by neil2468** [comment]
Hi. Sure, that’s fine by me.

**Comment by PsychoPunkSage** [comment]
Hi,
I'm planning to tackle this issue in 4 PRs. 1 PR for each pointer.
As it will allow me keep track of things that are completed and things that are left. Plus it will also motivate me to see some PRs made by me got merged.
Can I proceed in this way or should I do something else?
Thank you

**Comment by adrianbenavides** [comment]
@PsychoPunkSage Absolutely, that makes total sense, each point is worth its own PR.
